sklearn/gaussian_process/kernels.py

```diff
@@ -33,6 +33,21 @@
 from sklearn.externals.funcsigs import signature
 
 
+def _check_length_scale(X, length_scale):
+    if np.iterable(length_scale):
```

Comment:
Hmm. I thought about that but it creates a zero-dimensional array for a scalar. Are you sure that will not create any consequences?